ctru-rs/src/console.rs

```diff
@@ -1,27 +1,28 @@
+use std::cell::Ref;
 use std::default::Default;
 use std::marker::PhantomData;
 
 use ctru_sys::{consoleClear, consoleInit, consoleSelect, consoleSetWindow, PrintConsole};
 
-use crate::gfx::{Gfx, Screen};
+use crate::gfx::Screen;
 
-pub struct Console<'gfx> {
+pub struct Console<'screen> {
     context: Box<PrintConsole>,
-    _gfx: PhantomData<&'gfx ()>,
+    _screen: PhantomData<&'screen ()>,
 }
 
-impl<'gfx> Console<'gfx> {
+impl<'screen> Console<'screen> {
     /// Initialize a console on the chosen screen, overwriting whatever was on the screen
     /// previously (including other consoles). The new console is automatically selected for
     /// printing.
-    pub fn init(_gfx: &'gfx Gfx, screen: Screen) -> Self {
+    pub fn init(screen: Ref<'screen, impl Screen>) -> Self {
```

Comment:
This should be just be a reference, since it doesn't _need_ to be a reference-counted handle:
```suggestion
    pub fn init(screen: &'screen impl Screen) -> Self {
```
Also could use `dyn` here instead of `impl`, but it doesn't really matter.

Reply:
I believe you haven't looked at my code, since it's the second time you talked about an `Rc` implementation. This isn't reference counting, this is internal mutability. Reference counting would bring us to have multiple owners for the `Screen`, which isn't what we want. This is a `RefCell`, which is a smart pointer for a mutating object. It's used only for internal mutability inside `Gfx`, so that we may use the screens/gfx separately.

Also, without using a `Ref` object, the rules for `RefCell`'s borrowing aren't correctly applied. We may even have to store the reference, now that I think about it.

Reply:
Sorry, the types are named pretty similar and sometimes I mix them up. But `RefCell` does actually count references (just not in the way "reference-counted" implies):
https://doc.rust-lang.org/stable/src/core/cell.rs.html#687

The code as it was when I reviewed it though would have functioned the same if you used `Ref` or `&`. At that time, the `Ref` wasn't used, so the only functional thing happening was associating the `'screen` lifetime with the returned `Console<'screen>` instance. Which would work the same with `Ref` or `&`.

Either way the code has now been updated to hold the ref. But if we just use `Ref` then there's really no difference compared to using immutable references, with or without `PhantomData`. There's no internal mutability happening here, so removing the `RefCell` doesn't break anything - Yes, I tested this locally. Probably you meant to use `RefMut`? That would actually utilize the `RefCell` (letting you take mutable references to the fields of `Gfx` but still allowing `&self` calls to it).